libqtile/core/manager.py

```diff
@@ -408,14 +408,78 @@ def update_net_desktops(self) -> None:
         self.core.update_net_desktops(self.groups, index)
 
     def update_client_list(self) -> None:
-        """Updates the client stack list
+        """Update X11 client list
 
         This is needed for third party tasklists and drag and drop of tabs in
         chrome
         """
         windows = [wid for wid, c in self.windows_map.items() if c.group]
         self.core.update_client_list(windows)
 
+    def update_client_order(self) -> None:
+        """Update X11 client order
+        This is needed for third party tasklists and drag and drop of tabs in
+        chrome
+        """
+        windows = [wid for wid, c in self.windows_map.items() if c.group]
+        stack = self.conn.query_tree()
+        # TODO: there appear to be some windows (starting VirtualBox creates one) that might not be properly unmanaged
+        windows = sorted([w for w in windows if w in stack], key=lambda wid: stack.index(wid))
+        self.core.update_client_order(windows)
+
+    def change_layer(self, wid, up=True):
```

Comment:
Why adding this method to the manager rather than the Window object? Especially if its main argument is a window id.

Reply:
Since the logic shuffles all windows around, I wouldn't really expect it in the `Window`-class. It also requires access to `windows_map`.  I'll think about it some more.

Reply:
I'm still not sure about this. We need access to potentially all other windows to find our new place in the stack. Is that something I should be accessing from a `Window`? If you think so, I'll move the method.